runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -513,25 +513,40 @@ function LanguageTree:register_cbs(cbs)
   end
 end
 
+---Determine whether {range} is completely inside {region} (range(region)==range -> true).
+---@param region node or 6-tuple srow, scol, sbyte, erow, ecol, ebyte.
+---@param inner range, 4-tuple srow, scol, erow, ecol.
+---@return boolean
 ---@private
-local function tree_contains(tree, range)
-  local start_row, start_col, end_row, end_col = tree:root():range()
+local function region_contains(region, range)
+  local start_row, start_col, end_row, end_col
+  if type(region) == 'table' then
+    start_row, start_col, end_row, end_col = region[1], region[2], region[4], region[5]
+  else
+    start_row, start_col, end_row, end_col = region:range()
+  end
+
   local start_fits = start_row < range[1] or (start_row == range[1] and start_col <= range[2])
   local end_fits = end_row > range[3] or (end_row == range[3] and end_col >= range[4])
 
-  return start_fits and end_fits
+  if start_fits and end_fits then
+    return true
+  end
+
+  return false
```

Comment:
This change seems unnecessary and makes the code worse.

```lua
return condition
```
is much better than
```lua
if condition then
    return true
end
return false
```

Reply:
Oh agreed, no idea how I missed that.
Thanks :+1: